### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by SebastianMC** [SebastianMC/obsidian-custom-sort#44] [Ideas]
> [!NOTE]
>
> The Roadmap needs significant update and integration of the notes below plus new additions

Having used the custom-sort plugin for many weeks now, and having collected some valuable feedback, the following vision of roadmap clarifies in my mind:
- update the documentation to cover all implemented features
- knowing about the advent of new Obsidian API to easily manipulate (e.g. update) the YAML frontmatter of the notes, it is tempting to become more friendly to less technical users and expose a button on folder (context menu): _"Capture the list of items in the folder into a sorting specification note"_
  - this would eliminate the need to manually create `sortspec` (or equivalent) notes for those who struggle with YAML syntax details -> the YAML frontmatter would be generated automatically
  - manual rearrangement of lines in such note would result in changing of the sorting order
  - seems tempting.
- on top of the above UI extension, expose a set of additional context menu entries on File Explorer items:
  - move the item up
  - move the item down
  - select item for reordering .... move selection above this item
  - select item for reordering .... move selection below this item
  - move the item to the top
  - move the item to the bottom
  - the above entries in context menu would work on top of the automatically generated sorting spec note
  - overall, I'm not a big fan of mixing a fully UI-based drag & drop approach with specification-driven automatic ordering. Thus thinking towards a hybrid approach as outlined here, with the option to disable the UI context entries in configuration

**Comment by TheLukewarmYeti** [comment]
As someone with barely the technological literacy to stumble into installing the plugin, I would be ecstatic if there were some Layman-Friendly way to sort. I could probably get the hang of the first option, but I would be overjoyed to have the second option.

**Comment by SebastianMC** [comment]
Ideas from recent discussions:

- I need to give some more thought to the idea of 'integrated properties lists', suspecting some syntax conflicts between YAML and valid file names and the sorting specification syntax.

- Support for notes links is very tempting! Similarly to the idea of generating list of entries (folders and notes) in a folder via some context menu.

(from #106)

**Comment by SebastianMC** [comment]
In the [2.0.2 release of the plugin](https://github.com/SebastianMC/obsidian-custom-sort/releases/tag/2.0.2) I added support for drag and drop ordering of File Explorer items via Bookmarks integration: 
- detailed instructions available on [project wiki page: How to order items with drag and drop](https://github.com/SebastianMC/obsidian-custom-sort/wiki/How-to-order-items-in-File-Explorer-with-drag-and-drop%3F)
- feel free to post questions and report issues in the github home of the plugin: [Custom File Explorer Sorting](https://github.com/SebastianMC/obsidian-custom-sort)

The roadmap needs to be updated - probably some UI-related ideas became obsolete.
